brille/design/Proposed_integrated_syntax_Brill_SpinW_Horace.md

```diff
@@ -0,0 +1,275 @@
+
+We propose the following syntax for powder data handling in Horace, should it be implemented:
+
+```matlab
+proj = projaxes('powder');
+cut2 = h.cut_sqw(sqw_file, proj, [qmin, qstep, qmax], [emin, estep, emax]);
+```
+
+This syntax should retain pixel information just like the equivalent single crystal invocation.
+
+
+## Modelling the data
+
+We assume that [SpinW](https://spinw.org/) would be used to model the magnetic excitations.
+To set up and evaluate a model on single crystal data with current syntax in Matlab is:
+
+```matlab
+tri = sw_model('triAF', 1); 
+spinw_pars = {'hermit', false, 'formfact', true, 'resfun', 'gauss'}; 
+sim1 = disp2sqw_eval(cut1, @tri.horace_sqw, spinw_pars, fwhm);
+```
+
+In Python it is:
+
+```python
+tri = h.sw_model('triAF', 1)
+spinw_pars = ['hermit', False, 'formfact', True, 'resfun', 'gauss']
+sim1 = h.disp2sqw_eval(cut1, tri.horace_sqw, fwhm, *spinw_pars)
+```
+
+As per [ADR #9](../../documentation/adr/0009-brille-integration.md), Brille is to be integrated within SpinW,
+which would permit the Matlab syntax:
+
+```matlab
+tri = sw_model('triAF', 1); 
+spinw_pars = {'hermit', false, 'formfact', true, 'resfun', 'gauss', ...
+              'use_brille', true, 'node_fraction_volume', 1e-5}; 
+sim1 = disp2sqw_eval(cut1, @tri.horace_sqw, spinw_pars, fwhm);
+```
+
+And in Python:
+
+```python
+tri = h.sw_model('triAF', 1)
+spinw_pars = ['hermit', False, 'formfact', True, 'resfun', 'gauss',
+              'use_brille', true, 'node_fraction_volume', 1e-5]
+sim1 = h.disp2sqw_eval(cut1, tri.horace_sqw, fwhm, *spinw_pars)
+```
+
+Whilst these syntax currently work, their implementation is still in review and have yet to be fully tested or released.
+The Python syntax also assumes that the Horace python distribution (`pyHorace`) will include SpinW as well as Horace.
+This would enable both to use the same Matlab interpreter process, and make intercommunication much easier and faster.
+
+Again these syntax should work for powder data objects, possibly replacing `h.disp2sqw`
+with a (yet to be implemented) `m.disp2sqw` method of `MSlice`.
+
+
+## Fitting the data with resolution convolution
+
+Horace includes a fitting framework which uses a Levenberg-Marquardt minimizer.
+The following example uses Brille and the TobyFit Monte Carlo resolution convolution method:
+
+```matlab
+proj = projaxes([1 0 0], [0 1 0]);
+cut1 = cut_sqw(sqw_file, proj, [hmin hstep hmax], [kmin kstep kmax], [lmin lmax], [emin estep emax]);
+model = pcsmo(JF1, JA, JF2, JF3, Jperp, D);   % User SpinW model defined as a function
+model_pars = {'mat', {'JF1', 'JA', 'JF2', 'JF3', 'Jperp', 'D(3,3)'}, ...
+              'hermit', false, 'optmem', 0, 'useFast', true, 'formfact', true, ...
+              'resfun', 'gauss', 'coordtrans', diag([2 2 1 1]), ...
+              'use_brille', 'node_volume_fraction', 1e-5, 'Qtrans', diag(1./[4 4 1])};
+sample = IX_sample(true,[0,0,1],[0,1,0],'cuboid',[0.01,0.05,0.01]);
+maps = maps_instrument(ei, freq, 'S');
```

Comment:
the lattice parameters alatt and angdeg have now been moved from the generic run data to the sample - as they are sample-specific data. Once the sample object has been generated at line 182 you need to add
> sample.alatt = value
> sample.angdeg = value